resources/install.rb

```diff
@@ -24,37 +24,79 @@
 
 default_action :create
 
+def create_yum_repository
+  yum_repository 'telegraf' do
+    description 'InfluxDB Repository - RHEL \$releasever'
+    case node['platform']
+    when 'redhat'
+      baseurl  "#{node['telegraf']['package_url']}/rhel/\$releasever/\$basearch/stable"
+    when 'amazon'
+      baseurl  "#{node['telegraf']['package_url']}/centos/7/\$basearch/stable"
+    else
+      baseurl  "#{node['telegraf']['package_url']}/centos/\$releasever/\$basearch/stable"
+    end
+    gpgkey  "#{node['telegraf']['package_url']}/influxdb.key"
+    only_if { new_resource.include_repository }
+  end
+end
+
+def create_deb_repository
+  package 'apt-transport-https' do
+    only_if { new_resource.include_repository }
+  end
+
+  case node['kernel']['machine']
+  when 'x86_64'
+    telegraf_arch = 'amd64'
+  when 'i686', 'i386'
+    telegraf_arch = 'i386'
+  when 'armv7l', 'armv6l'
+    telegraf_arch = 'armhf'
+  when 'armv5l'
```

Comment:
nice

Reply:
i can hardly take much credit for this - the upstream people have been doing stuff :D